measures/BuildExistingModel/measure.rb

```diff
@@ -257,6 +292,82 @@ def run(model, runner, user_arguments)
     end
     measures['BuildResidentialHPXML'][0]['simulation_control_run_period_calendar_year'] = args['simulation_control_run_period_calendar_year'].get if args['simulation_control_run_period_calendar_year'].is_initialized
 
+    # Emissions
+    if args['emissions_scenario_names'].is_initialized
+      ba_to_gea_csv_path = File.join(resources_dir, 'ba_to_gea.csv')
+      ba_to_gea_csv = CSV.open(ba_to_gea_csv_path, headers: true)
+
+      gea = nil
+      CSV.foreach(ba_to_gea_csv_path, headers: true) do |row|
+        next if row[0] != "p#{bldg_data['REEDS Balancing Area']}"
+
+        gea = row[1]
+      end
+
+      if gea.nil?
+        runner.registerError("Could not find balancing area 'p#{bldg_data['REEDS Balancing Area']}' in #{ba_to_gea_csv_path}.")
+        return false
+      end
+
+      emissions_electricity_filepaths = []
+      scenarios = args['emissions_electricity_folders'].get.split(',')
+      scenarios.each do |scenario|
+        scenario = File.join(resources_dir, scenario)
+        if not File.exist? scenario
+          runner.registerError("CO2 emissions scenario folder '#{scenario}' does not exist.")
+          return false
+        end
+
+        Dir["#{scenario}/*.csv"].each do |filepath|
+          emissions_electricity_filepaths << filepath if filepath.include?(gea)
+        end
+      end
+
+      emissions_scenario_names = args['emissions_scenario_names'].get
+      emissions_types = args['emissions_types'].get
+      emissions_electricity_filepaths = emissions_electricity_filepaths.join(',')
+      emissions_electricity_units = ([HPXML::EmissionsScenario::UnitsLbPerMWh] * scenarios.size).join(',')
```

Comment:
This would be better as kg/MWh, as that is what comes from cambium. Or we could expose them as inputs in the yml file.

Reply:
Let's hardcode to kg/MWh (electricity) and lb/MBtu (fuel). In the yml update the comment to express "million Btu".